src/core/ext/filters/client_channel/http_connect_handshaker.cc

```diff
@@ -255,15 +265,16 @@ static void http_connect_handshaker_do_handshake(
   const grpc_arg* arg =
       grpc_channel_args_find(args->args, GRPC_ARG_HTTP_CONNECT_SERVER);
   if (arg == nullptr) {
```

Comment:
I'm fine with this change, but instead of moving the shutdown code to its own function, please just change this line to say `arg == nullptr || arg->type != GRPC_ARG_STRING`.